src/ccache.cpp

```diff
@@ -2766,95 +2621,88 @@ cc_process_args(struct args* args,
     if (str_startswith(argv[i], "-MQ") || str_startswith(argv[i], "-MT")) {
       dependency_target_specified = true;
 
-      char* relpath;
       if (strlen(argv[i]) == 3) {
         // -MQ arg or -MT arg
         if (i == argc - 1) {
           cc_log("Missing argument to %s", argv[i]);
           stats_update(STATS_ARGS);
-          result = false;
-          goto out;
+          return false;
         }
         args_add(dep_args, argv[i]);
-        relpath = make_relative_path(x_strdup(argv[i + 1]));
-        args_add(dep_args, relpath);
-        free(relpath);
+        std::string relpath = make_relative_path(argv[i + 1]);
+        args_add(dep_args, relpath.c_str());
         i++;
       } else {
         char* arg_opt = x_strndup(argv[i], 3);
-        relpath = make_relative_path(x_strdup(argv[i] + 3));
-        char* option = format("%s%s", arg_opt, relpath);
+        std::string relpath = make_relative_path(argv[i] + 3);
+        char* option = format("%s%s", arg_opt, relpath.c_str());
         args_add(dep_args, option);
         free(arg_opt);
-        free(relpath);
         free(option);
       }
       continue;
     }
     if (str_eq(argv[i], "-fprofile-arcs")) {
-      profile_arcs = true;
+      args_info.profile_arcs = true;
       args_add(common_args, argv[i]);
       continue;
     }
     if (str_eq(argv[i], "-ftest-coverage")) {
-      generating_coverage = true;
+      args_info.generating_coverage = true;
       args_add(common_args, argv[i]);
       continue;
     }
     if (str_eq(argv[i], "-fstack-usage")) {
-      generating_stackusage = true;
+      args_info.generating_stackusage = true;
       args_add(common_args, argv[i]);
       continue;
     }
     if (str_eq(argv[i], "--coverage")      // = -fprofile-arcs -ftest-coverage
         || str_eq(argv[i], "-coverage")) { // Undocumented but still works.
-      profile_arcs = true;
-      generating_coverage = true;
+      args_info.profile_arcs = true;
+      args_info.generating_coverage = true;
       args_add(common_args, argv[i]);
       continue;
     }
     if (str_startswith(argv[i], "-fprofile-dir=")) {
-      profile_dir = x_strdup(argv[i] + 14);
+      args_info.profile_dir = from_owned_cstr(x_strdup(argv[i] + 14));
```

Comment:
Isn't this the same?
```suggestion
      args_info.profile_dir = from_cstr(argv[i] + 14);
```